blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,666 @@
+be considered valid in those same contexts - whatever they may be.
+
+```typescript
+interface Server {
+
+	cachegroup?:  string | null;
+	cachegroupId: number;
+	cdnId:        number;
+	cdnName?:     string | null;
+	domainName:   string;
+	guid?:        string | null;
+	hostName:     string;
+	httpsPort?:   number | null;
+	id?:          number;
+
+	// ILO things aren't being moved in because they also require a
+	// username/password that aren't used by any other interface type.
+	iloIpAddress?: string | null;
+	iloIpGateway?: string | null;
+	iloIpNetmask?: string | null;
+	iloPassword?:  string | null;
+	iloUsername?:  string | null;
+
+	/**
+	 * New field containing additional interfaces.
+	 * It is illegal for a server to not have at least one interface.
+	 */
+	interfaces: Array<Interface> & {0: Interface};
+	lastUpdated?:  string;
+
+	// notice not here: mgmtIpAddress, mgmtIpGateway, mgmtIpNetmask
+	// these will be moved into the interfaces property
+
+	offlineReason:   string;
+	physLocation?:   string | null;
+	physLocationId:  number;
+	profile?:        string | null;
+	profileDesc?:    string | null;
+	profileId:       number;
+	revalPending?:   boolean;
+	rack?:           string | null;
+	routerHostName?: string | null;
+	routerPortName?: string | null;
+	status?:         string | null;
+	statusId:        number;
+	tcpPort:         number;
+	type?:           string | null;
+	typeId:          number;
+	updPending?:     boolean;
+	xmppId?:         string | null;
+	xmppPassword?:   string | null;
+}
+```
+
+`/monitoring` payloads and CDN Snapshots are purposely un-versioned and
+un-modeled, so this section will make no attempts to define the model changes to
+those objects.
+
+## Impacted Components
+The only components that should need to be modified to accommodate this feature
+are Traffic Ops, Traffic Portal, and Traffic Monitor.
+
+### Traffic Ops Impact
+
+#### Database
+A new table will need to be introduced to contain interface information:
+`interface`, which is described below.
+```text
+              Table "public.interface"
+ Column  |  Type   | Collation | Nullable | Default
+---------+---------+-----------+----------+---------
+ name    | text    |           | not null |
+ server  | bigint  |           | not null |
+ monitor | boolean |           | not null |
+ mtu     | integer |           |          | 1500
+Indexes:
+    "interface_pkey" PRIMARY KEY, btree (name, server)
+Foreign-key constraints:
+    "interface_server_fkey" FOREIGN KEY (server) REFERENCES server(id)
+```
+Another new table will need to be created to contain address information, which
+is described below.
+```text
+                 Table "public.ip_address"
+     Column     |  Type   | Collation | Nullable | Default
+----------------+---------+-----------+----------+---------
+ address        | inet    |           | not null |
+ gateway        | inet    |           | not null |
+ serviceaddress | boolean |           | not null |
```

Comment:
camel case?

Reply:
snake_case is what we typically use, because PostgreSQL (SQL standard?) case-folds everything you don't mark as an explicit identifier with double-quotes (<kbd>"</kbd>) to lowercase. Which is what happened here. That's just an example name, though, the only thing that's meant to specify is table structure, the column and table names are all arbitrary (provided you know what they're for).

Reply:
gotcha and sorry, i meant snake. got my snakes and camels screwed up. :)